addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -18,10 +18,230 @@ var globalSelector = {
     is: () => false,
 };
 
+/**
+ * Mixin that provides smooth scroll behaviour.
+ *
+ * Call setDraggable in order to set the behaviour.
+ * Call unsetDraggable to ensure proper resource clearing.
+ */
+const SmoothOnDragMixin = {
```

Comment:
Instead of a mixin which is hard to extend, why not creating a real Class (Odoo web.Class) which is instantiated by the objects who needs it?

Instead of:
```javascript
const Mixin = {
    a: , //...
    b: , //...
};
const Editor = Widget.extend(Mixin, { // Potential conflict of mixin functions and mines
    a: function () {
        this.setDraggable(/* ... */); // I have to know this comes from the mixin
    },
});
return {
    Editor: Editor,
    Mixin: Mixin, // Can be exposed like this but hard to extend
};
```
Doing:
```javascript
const Component = Class.extend({
    a: , //...
    b: , //...
});
const Editor = Widget.extend({
    a: function () {
        const component = new Component(/* ... */); // No name conflict + explicit
    },
});
return {
    Editor: Editor,
    Component: Component, // Easy to extend by external code through extend/include
};
```
What do you think?